tests/drivers/uart/uart_async_api/src/test_uart.h

```diff
@@ -21,6 +21,8 @@
 
 #if DT_NODE_EXISTS(DT_NODELABEL(dut))
 #define UART_NODE DT_NODELABEL(dut)
+#elif defined(CONFIG_SOC_SERIES_ESP32C3)
```

Comment:
ESP32C3 already have `dut` node in its overlay, right?

Reply:
yes, but not all the ESP32C3 boards has its overlay

Reply:
Oh right. Then I think it is better to include an overlay for the target board with `dut` node if it wants to run this test. If you take a look at the file history, you will see that used to be a lot of `#ifdefs` defining the UART node according to the SOC until `dut` node was introduced. But I don't think including any overlay is in the scope of this PR.

Reply:
maybe we can update this files in upgoing PR, what do you think?